pkg/kubelet/cm/dra/state/state_checkpoint.go

```diff
@@ -0,0 +1,115 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package state
+
+import (
+	"fmt"
+	"sync"
+
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/kubernetes/pkg/kubelet/checkpointmanager"
+	"k8s.io/kubernetes/pkg/kubelet/checkpointmanager/errors"
+)
+
+var _ CheckpointState = &stateCheckpoint{}
+
+// CheckpointState interface provides to get and store state
+type CheckpointState interface {
+	GetOrCreate() (ClaimInfoStateList, error)
+	Store(ClaimInfoStateList) error
+}
+
+// ClaimInfoState is used to store claim info state in a checkpoint
+type ClaimInfoState struct {
+	// Name of the DRA driver
+	DriverName string
+
+	// ClaimUID is an UID of the resource claim
+	ClaimUID types.UID
+
+	// ClaimName is a name of the resource claim
+	ClaimName string
+
+	// Namespace is a claim namespace
+	Namespace string
+
+	// PodUIDs is a set of pod UIDs that reference a resource
+	PodUIDs sets.Set[string]
+
+	// CdiDevices is a list of CDI devices returned by the
+	// GRPC API call NodePrepareResource
+	CdiDevices []string
```

Comment:
Something that came out of reviewing:

To future proof this, let's store it as a map, where the key is the name of the kubelet plugin. For now there will only be one entry (whose name is the same as the `DriverName`), but in the future there will be multiple entries.

Reply:
Actually. I will fold this into my changes for https://github.com/kubernetes/kubernetes/pull/116332 which is where it becomes relevant anyway. If for some reason it becomes clear that that PR wont make it, we can pull it out as a standalone change so we don't have to change the checkpointing format in 1.28.

Reply:
Would it make sense to rename it to CDIDevices as we did for CRI?

Reply:
Yes, but I will do this in a follow-up.